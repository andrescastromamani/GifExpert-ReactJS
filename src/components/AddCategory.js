import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');
    const handleAddCategory = (e) => {
        setInputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 0) {
            setCategories(cats => [inputValue, ...cats])
            setInputValue('');
        }
    }
    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                placeholder={inputValue}
                value={inputValue}
                onChange={handleAddCategory} />
        </form>
    )
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}