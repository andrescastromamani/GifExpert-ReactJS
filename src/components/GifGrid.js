import React, { useEffect, useState } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
//import { getGifs } from '../helpers/getGifs'
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
    /*const [images, setImages] = useState([])
    useEffect(() => {
        getGifs(category).then(setImages)
    }, [category])*/
    const { loading, data } = useFetchGifs();
    console.log(data);
    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                {
                    loading ? 'Loading....' : 'Listo'
                }
                {/*
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))*/
                }
            </div>
        </>
    )
}