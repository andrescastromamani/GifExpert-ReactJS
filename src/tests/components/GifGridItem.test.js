import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('GifGridItem', () => {
    const title = 'Title'
    const url = 'https://media1.giphy.com/media/l0HlW7JqkvXy/200w.gif'
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('show the component GifGridItem', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('verify if content paragraph with title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('verify src and alt', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('verify if contain the class animate__bounce ', () => {
        const div = wrapper.find('div');
        //console.log(div.prop('className'));
        expect(div.hasClass('animate__bounce')).toBe(true);
    })
})