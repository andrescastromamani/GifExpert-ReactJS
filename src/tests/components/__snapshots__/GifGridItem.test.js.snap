// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem show the component GifGridItem 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Title"
    src="https://media1.giphy.com/media/l0HlW7JqkvXy/200w.gif"
  />
  <p>
    Title
  </p>
</div>
`;
